src/components/Models/MultipleAnimationModel/MultipleAnimationModel.component.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { FC, useEffect, useRef, useState } from 'react';
+import { useFrame, useLoader } from '@react-three/fiber';
+import { AnimationAction, AnimationClip, AnimationMixer, Group } from 'three';
+import { GLTFLoader } from 'three-stdlib';
+
+import { Model } from 'src/components/Models/Model';
+import { BaseModelProps } from 'src/types';
+
+export interface MultipleAnimationModelProps extends BaseModelProps {
+  modelSrc: string | Blob;
+  animations: Record<string, string>;
+  activeAnimation: string;
+  scale?: number;
+}
+
+export const MultipleAnimationModel: FC<MultipleAnimationModelProps> = ({
+  modelSrc,
+  animations,
+  activeAnimation,
+  scale = 1,
+  onLoaded,
+  bloom
+}) => {
+  const groupRef = useRef<Group>(null);
+  const mixerRef = useRef<AnimationMixer | null>(null);
+  const [loadedAnimations, setLoadedAnimations] = useState<Record<string, AnimationClip>>({});
+  const [activeAction, setActiveAction] = useState<AnimationAction | null>(null);
+
+  const { scene } = useLoader(GLTFLoader, String(modelSrc));
+
+  useEffect(() => {
+    if (scene && groupRef.current) {
+      groupRef.current.add(scene);
+
+      mixerRef.current = new AnimationMixer(scene);
+    }
+
+    return () => {
+      mixerRef.current?.stopAllAction();
+    };
+  }, [scene]);
+
+  useEffect(() => {
+    const loader = new GLTFLoader();
+    Object.keys(animations).forEach((name) => {
+      loader.load(animations[name], (gltf) => {
+        const newClip = gltf.animations[0];
+        setLoadedAnimations((prev) => ({
+          ...prev,
+          [name]: newClip
+        }));
+      });
+    });
+  }, [animations]);
+
+  useEffect(() => {
+    const mixer = mixerRef.current;
+    const newClip = loadedAnimations[activeAnimation];
+
+    if (!newClip || !mixer) return;
+
+    const newAction = mixer.clipAction(newClip);
+    const prevAction = activeAction;
+
+    if (prevAction) {
+      newAction.reset();
+      newAction.crossFadeFrom(prevAction, 0.5, true);
+    }
+
+    newAction.play();
+    setActiveAction(newAction);
+  }, [activeAnimation, loadedAnimations, activeAction]);
```

Comment:
If are using emotion, [and nodes](https://github.com/readyplayerme/visage/blob/main/src/components/Models/AnimationModel/AnimationModel.component.tsx#L42), then we can add nodes.Armature to this dependancy.

```suggestion
  }, [activeAnimation, loadedAnimations, activeAction, nodes.Armature]);
```

At the moment, if you change the model url then the animation won't apply to the new loaded model.

Reply:
yep, will add emotions also

Reply:
<img width="643" alt="image" src="https://github.com/user-attachments/assets/6f1271ba-b9b4-4b64-9b65-a5a146b11557">
Fixed